core/services/vrf/v2/coordinator_v2x_interface.go

```diff
@@ -545,7 +546,30 @@ func (c *coordinatorV2_X) RequestRandomWords(opts *bind.TransactOpts, keyHash [3
 	if c.v2 != nil {
 		return c.v2.RequestRandomWords(opts, keyHash, subId, requestConfirmations, callbackGasLimit, numWords)
 	}
-	return c.v2plus.RequestRandomWords(opts, keyHash, subId, requestConfirmations, callbackGasLimit, numWords, payInEth)
+	extraArgs, err := GetExtraArgsV1(payInEth)
+	if err != nil {
+		return nil, err
+	}
+	req := vrf_coordinator_v2plus.VRFV2PlusClientRandomWordsRequest{
+		KeyHash:              keyHash,
+		SubId:                subId,
+		RequestConfirmations: requestConfirmations,
+		CallbackGasLimit:     callbackGasLimit,
+		NumWords:             numWords,
+		ExtraArgs:            extraArgs,
+	}
+	return c.v2plus.RequestRandomWords(opts, req)
+}
+
+func GetExtraArgsV1(nativePayment bool) ([]byte, error) {
+	extraArgsV1Tag := []byte{0x92, 0xfd, 0x13, 0x38}
```

Comment:
Is there a better way to represent these rather than magic bytes?

Reply:
Perhaps you can do (pseudocode)

```
h := keccak256("VRF ExtraArgsV1")
tag := h[:4]
```

Maybe you can make it a one-liner and stick it in the `var` declarations. Or perhaps the best would be if this is fetched directly from the Go wrappers, but only if it can be declared as a static constant that can be accessed without needing to instantiate the wrapper, though I doubt that's possible.

Reply:
yeah that sounds like the better thing to do 👍